src/xrc/src/forex.rs

```diff
@@ -277,16 +277,18 @@ impl ForexRateStore {
         quote_asset: &str,
     ) -> Result<QueriedExchangeRate, GetForexRateError> {
         // Normalize timestamp to the beginning of the day.
-        let mut timestamp = (requested_timestamp / SECONDS_PER_DAY) * SECONDS_PER_DAY;
+        let mut requested_timestamp = (requested_timestamp / SECONDS_PER_DAY) * SECONDS_PER_DAY;
 
+        // If today's date is requested, and the day is not over anywhere on Earth, use yesterday's date
+        // Get the normalized timestamp for yesterday.
         if !cfg!(feature = "disable-forex-timezone-offset") {
             // If today's date is requested, and the day is not over anywhere on Earth, use yesterday's date
             // Get the normalized timestamp for yesterday.
-            let yesterday = (current_timestamp as i64 + TIMEZONE_AOE_SHIFT_SECONDS) as u64
-                / SECONDS_PER_DAY
-                * SECONDS_PER_DAY;
-            if timestamp > SECONDS_PER_DAY && yesterday == timestamp {
-                timestamp -= SECONDS_PER_DAY;
+            // If today's date is requested, and the day is not over anywhere on Earth, use yesterday's date
+            // Get the normalized timestamp for yesterday.
+            let aoe_end = requested_timestamp.saturating_add(TIMEZONE_AOE_SHIFT_SECONDS);
```

Comment:
I keep getting confused here, as when seeing `aoe_end` I think it means "today's end AOE", but it actually means "requested timestamp's day end AOE". So maybe rename it to `requested_day_aoe_end`?
(I know it's long and ugly, but may prevent future confusion)